src/frontends/android/app/src/main/java/org/strongswan/android/security/LocalKeystore.java

```diff
@@ -204,10 +204,12 @@ public String generateId() {
 
     public boolean removePkcs12AndCaCertificate(String certificateId) {
         try {
-            File[] certificateFiles = StrongSwanApplication.getContext().getFilesDir().listFiles();
-            for (File certificate : certificateFiles) {
-                if (certificate.getName().contains(certificateId)) {
-                    certificate.delete();
+            if(certificateId != null) {
```

Comment:
Its public method so just in case better is !TextUtil.isEmpty instead of only null checking, I bet string.contains("") returns true.